src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -107,10 +109,25 @@ private ZipFileProcessor processZipFileEntry(
             ZipFileProcessor zipFileProcessor = new ZipFileProcessor(containerName, zipFilename);
             zipFileProcessor.process(zis);
 
-            Envelope envelope = envelopeProcessor.processEnvelope(zipFileProcessor.getMetadata(), containerName);
-            zipFileProcessor.setEnvelope(envelope);
+            Envelope envelope = envelopeProcessor.parseEnvelope(zipFileProcessor.getMetadata());
+            envelope.setContainer(containerName);
 
-            return zipFileProcessor;
+            Optional<Envelope> optionalEnvelope = envelopeProcessor.hasEnvelopeFailedToUploadBefore(envelope);
+
+            if (optionalEnvelope.isPresent()) {
+                throw new PreviouslyFailedToUploadException(
```

Comment:
Do we need to throw an exception ? we can just log and skip the record

Reply:
Error wrapper does that so you don't worry and continue the work